cram_core/cram_occasions_events/src/tasks.lisp

```diff
@@ -44,11 +44,344 @@
+    (bound ?fluent)
+    (bound ?top-level-name)
+    (top-level-episode-knowledge ?top-level-name ?episode)
+    (lisp-pred identity ?fluent)
+    (lisp-fun cpl-impl:name ?fluent ?fluent-name)
+    (lisp-fun cet:episode-knowledge-fluent-durations ?fluent-name ?episode ?durations)
+    (member (?value . ?duration) ?durations)
+    (cram-occasions-events:duration-includes ?duration ?time))
+
+  (<- (coe:holds (task-status ?task ?status) ?top-level-name ?time)
+    (bound ?top-level-name)
+    (task-status-fluent ?task ?status-fluent)
+    (coe:holds (fluent-value ?status-fluent ?status) ?top-level-name ?time))
+
+  ;; task times
+  (<- (task-created-at ?top-level-name ?task ?time)
+    (bound ?top-level-name)
+    (bound ?task)
+    (coe:holds (task-status ?task :created) ?top-level-name (coe:at ?time)))
+
+  (<- (task-started-at ?top-level-name ?task ?time)
+    (bound ?top-level-name)
+    (bound ?task)
+    ;; (task ?task)
+    (bagof ?time
+           (coe:holds (task-status ?task :running) ?top-level-name (coe:at ?time))
+           ?times)
+    (sort ?times < (?time . ?_)))
+
+  (<- (task-ended-at ?top-level-name ?task ?time)
+    (bound ?top-level-name)
+    (bound ?task)
+    ;; (task ?task)
+    (member ?status (:succeeded :failed :evaporated))
+    (coe:holds (task-status ?task ?status) ?top-level-name (coe:at ?time)))
+
+  ;; task next and previous sibling
+  (<- (task-next-sibling ?top-level-name ?task ?next-task)
+    (bound ?top-level-name)
+    (bound ?task)
+    (bagof ?sibling (task-sibling ?task ?sibling) ?siblings)
+    (member ?next-task ?siblings)
+    (task-created-at ?top-level-name ?task ?created-time-task)
+    (task-created-at ?top-level-name ?next-task ?created-time-next-task)
+    (<= ?created-time-task ?created-time-next-task)
+    (forall (and
+             (member ?other-next-task ?siblings)
+             (not (== ?next-task ?other-next-task))
+             (task-created-at ?top-level-name ?other-next-task ?created-time-other-next-task)
+             (<= ?created-time-task ?created-time-other-next-task))
+            (<= ?created-time-next-task ?created-time-other-next-task)))
+
+  (<- (task-previous-sibling ?top-level-name ?task ?next-task)
+    (bound ?top-level-name)
+    (bound ?task)
+    (bagof ?sibling (task-sibling ?task ?sibling) ?siblings)
+    (member ?next-task ?siblings)
+    (task-created-at ?top-level-name ?task ?created-time-task)
+    (task-created-at ?top-level-name ?next-task ?created-time-next-task)
+    (>= ?created-time-task ?created-time-next-task)
+    (forall (and
+             (member ?other-next-task ?siblings)
+             (not (== ?next-task ?other-next-task))
+             (task-created-at ?top-level-name ?other-next-task ?created-time-other-next-task)
+             (>= ?created-time-task ?created-time-other-next-task))
+            (>= ?created-time-next-task ?created-time-other-next-task)))
+
+
+  #+still-in-cram-pr2-projection-reasoning-bc-depends-on-cram-executive
```

Comment:
``cram_pr2_projection_reasoning`` is really not the right package for this.
These predicates don't depend on PR2 or projection...
Unfortunately, putting them into ``cram_occasions_events`` and adding dependency on ``cram_executive`` is impossible, as that would create a circular dependency, because ``cram_executive`` depends on ``cram_occasions_events`` and that cannot be changed. I think the next best place to put these predicates is then ``cram_plan_occasions_events``.